npm/src/s3upload/index.ts

```diff
@@ -0,0 +1,78 @@
+import S3 from "aws-sdk/clients/s3"
+import {
+  IProgressInformation,
+  TProgressFunction
+} from "@nationalarchives/file-information"
+export interface ITdrFile {
+  fileId: string
+  file: File
+}
+
+interface IFileProgressInfo {
+  processedChunks: number
+  totalChunks: number
+  totalFiles: number
+}
+
+type TdrS3 = Pick<S3, "upload">
+export class S3Upload {
+  s3: TdrS3
+  identityId: string
+  constructor(identityId: string) {
+    this.s3 = new S3()
+    this.identityId = identityId
+  }
+  private uploadSingleFile: (
+    tdrFile: ITdrFile,
+    callback: TProgressFunction,
+    progressInfo: IFileProgressInfo
+  ) => Promise<S3.ManagedUpload.SendData> = (
+    tdrFile,
+    callback,
+    progressInfo
+  ) => {
+    const { file, fileId } = tdrFile
+    console.log(`${this.identityId}/${fileId}`)
+    const progress: S3.ManagedUpload = this.s3.upload({
+      Key: `${this.identityId}/${fileId}`,
+      Body: file,
+      Bucket: "tdr-upload-files-intg"
```

Comment:
Should the bucket name be hardcoded? Would be different depending on the environment

Reply:
Good spot, I've added the stage variable and used it here.